security/interceptor/security-interceptor-anonymous/src/main/java/org/codice/ddf/security/interceptor/AnonymousInterceptor.java

```diff
@@ -376,18 +388,23 @@ public void handleMessage(SoapMessage message) throws Fault {
     }
 
     private void createSecurityToken(SoapVersion version, SOAPFactory soapFactory, SOAPElement securityHeader) {
-        AnonymousAuthenticationToken token = new AnonymousAuthenticationToken(BaseAuthenticationToken.DEFAULT_REALM);
-
-        //synchronize the step of requesting the assertion, it is not thread safe
-        Subject subject = null;
-        synchronized (lock) {
-            try {
-                subject = securityManager.getSubject(token);
-            } catch (SecurityServiceException sse) {
-                LOGGER.warn("Unable to request subject for anonymous user.", sse);
+        Subject subject = anonymousSubject;
+        if (tokenAboutToExpire()) {
+            AnonymousAuthenticationToken token = new AnonymousAuthenticationToken(BaseAuthenticationToken.DEFAULT_REALM);
+            LOGGER.debug("Getting new Anonymous user token");
+            //synchronize the step of requesting the assertion, it is not thread safe
+            synchronized (lock) {
+                try {
+                    subject = securityManager.getSubject(token);
+                    anonymousSubject = subject;
+                } catch (SecurityServiceException sse) {
+                    LOGGER.warn("Unable to request subject for anonymous user.", sse);
+                }
             }
-
+        } else {
+            LOGGER.debug("Using cached Anonymous user token");
         }
+
```

Comment:
If anonymousSubject is null and securityManager.getSubject throws an exception, then it is possible for the value to still be null.  This could happen for example if there is a connection issue with the STS.